public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx

```diff
@@ -115,13 +128,148 @@ export const AnomalyDetailsChart = React.memo(
     });
     const [zoomedAnomalies, setZoomedAnomalies] = useState<any[]>([]);
 
+    const [aggregatedAnomalies, setAggregatedAnomalies] = useState<any[]>([]);
+    const [selectedAggId, setSelectedAggId] = useState<ANOMALY_AGG>(
+      ANOMALY_AGG.RAW
+    );
+    const [disabledAggsMap, setDisabledAggsMap] = useState<
+      { [key in ANOMALY_AGG]: boolean }
+    >({
+      raw: false,
+      day: true,
+      week: true,
+      month: true,
+    });
+
     const DEFAULT_DATE_PICKER_RANGE = {
       start: moment().subtract(7, 'days').valueOf(),
       end: moment().valueOf(),
     };
 
+    const taskId = get(props, 'detector.taskId');
+    const taskState = get(props, 'detector.taskState');
+
+    const isRequestingAnomalyResults = useSelector(
+      (state: AppState) => state.anomalyResults.requesting
+    );
+
+    const getAggregatedAnomalies = async () => {
+      const anomalyDataRangeQuery = getAnomalyDataRangeQuery(
+        zoomRange.startDate,
+        zoomRange.endDate,
+        taskId
+      );
+      dispatch(searchResults(anomalyDataRangeQuery))
+        .then((response: any) => {
+          // Only retrieve buckets that are in the anomaly results range. This is so
+          // we don't show aggregate results for where there is no data at all
```

Comment:
It's possible that some AD result with 0 grade not persisted in index if writing pressure was high. 
If there is no data for some detection intervals in the anomaly results range, will we fill these intervals as 0?

Reply:
No. This first query is trimming down the selected date range, to only aggregate in the range that has results persisted in the index. This is done for a few reasons:
1. Optimize the query to limit the number of needed aggregations
2. Prevent showing weird results where the aggregate values would all be 0, when the root cause is simply that there is no data, and not non-anomalous data.

Note that the optimization described above is only for setting the beginning and end times. If there is missing data _within_ those time bounds, it will essentially be counted as 0, since the aggregation query is using the date histogram aggregation and bucketing aggregated results in different time chunks. Since we only are aggregating over max grade for now, this won't affect these results.

Reply:
Does that mean when cx select a zoom range, you would show that zoom range regardless there are data or not.  In the zoom range, the grade would be 0 when there is no results or the max grade are actually zero.  Is that correct?

Reply:
No, we will actually show no data point at all if there was no results, rather than 0. If the max grade is actually 0 (existing data points) we show a 0 data point.

Example:
`-` : data point (anomalous or not)
`x`: no data point

Given:
`x x x - - - - - - - x`

Result output:
`x x x .3 .1 0 .9 0 0 .6 x`

In other words, we do an initial query to only aggregate where there is existing data based on a cx-selected zoom range, for the 2 reasons mentioned above. Where there is no anomaly results, there is also no aggregated results. The exception to this, is if there is missing data in between data points, it may show as 0 even if there is missing results, since there is no easy way to evaluate each bucket when doing the actual aggregation in the following query.

Reply:
could you point me the code how you differentiate no data and zero grade results? In anomalyResultUtils.parseHistoricalAggregatedAnomalies:

```
anomalies.push({
      anomalyGrade: maxGrade !== null ? maxGrade : 0,
```

Since the way you structure the query (see https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/16#discussion_r667160090), maxGrade = null due to missing data or zero grade, right?

Reply:
You are correct, if there is missing data within data points it sets the grade as 0. The first query is simply an optimization to trim down the overall search space for the second aggregation query. I edited the example above to illustrate that

Reply:
I am confused.  You agreed that you cannot precisely differentiate no data and zero grade. But you still show "no data" to cx. Could you explain more?

Reply:
Missing data is handled 2 different ways:
1. If there is no data at the beginning or end of the zoom range, we don't aggregate or retrieve results for those ranges, which means we have no data points that will show. This is the purpose of the first query. For example, say the zoom range is 7 days, but the detector has only been running for the last 2 days, then we will only aggregate and show results for the last 2 days, and there will be no data for the other 5 days.
2. If there is missing data _in between_ existing data, then the buckets may possibly return `null` for max grade, which we convert to a max grade of 0.

Reply:
It's different in that the RT case for missing data is about a feature data point not existing in an interval window. In this scenario, we are aggregating over a user-specified time range, where there is potentially a lot of data points in one bucket (consider a one-month time range with 1-minute intervals). So, we would only show a 'missing data' in red if every data point (anomalous or non-anomalous) in that interval was missing.

While this is possible, it would require aggregating over all non-anomalous data points as well which may make the query a lot slower, especially since the primary use case for this is chunking large stretches of historical data. Also, this could add some user confusion. For example, if data is missing for one whole day, users may see a missing data annotation when doing daily aggregations, but the warning may go away when choosing a coarser-grained aggregation, like month-long aggregations.